inventory-service/src/main/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceImpl.java

```diff
@@ -558,6 +558,21 @@ public Mono<ProductResponseDTO> addSupplyToInventory(Mono<ProductRequestDTO> pro
 
     }
 
+    @Override
+    public Mono<ProductResponseDTO> consumeProduct(String inventoryId, String productId) {
+        return productRepository.findProductByInventoryIdAndProductId(inventoryId, productId)
+                .flatMap(product -> {
+                    if (product.getProductQuantity() - 1 < 0) {
+                        return Mono.error(new InvalidInputException("Not enough stock to consume."));
+                    } else {
+                        product.setProductQuantity(product.getProductQuantity() - 1);
+                        return productRepository.save(product)
+                                .map(EntityDTOUtil::toProductResponseDTO);
+                    }
+                })
+                .switchIfEmpty(Mono.error(new NotFoundException("Product not found with id: " + productId)));
+    }
```

Comment:
The logic for checking whether a product quantity is 0 or bigger than 0 is simple and well implemented.

Reply:
Very nice and well done !